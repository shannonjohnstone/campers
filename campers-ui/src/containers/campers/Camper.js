import React from 'react';
import PropTypes from 'prop-types';

const Camper = ({ camper }) => (
  <tr>
    <td>{camper.rank}</td>
    <td>
      <img src={camper.avatar} alt={camper.username} />
    </td>
    <td>{camper.username}</td>
    <td>{camper.monthly_total}</td>
    <td>{camper.total}</td>
  </tr>
);

Camper.propTypes = {
  camper: PropTypes.shape({
    rank: PropTypes.number.isRequired,
    avatar: PropTypes.string.isRequired,
    username: PropTypes.string.isRequired,
    monthly_total: PropTypes.number.isRequired,
    total: PropTypes.number.isRequired,
  }),
};

export default Camper;
