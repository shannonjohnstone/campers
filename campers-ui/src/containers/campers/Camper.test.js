import React from 'react';
import Camper from './Camper';
import { render } from 'react-testing-library';

const mockCamper = {
  id: '5c79ef86802e9c20cc0d52ce',
  img: 'https://source.unsplash.com/random/32x32',
  username: 'Luann Ballard',
  avatar:
    'https://www.tinygraphs.com/squares/859?theme=bythepool&numcolors=1&size=32&fmt=svg',
  monthly_total: 55,
  total: 3543,
  rank: 1,
};

describe('Camper component', () => {
  test('Component snapshot', () => {
    const { container } = render(
      <table>
        <tbody>
          <Camper camper={mockCamper} />
        </tbody>
      </table>,
    );
    expect(container).toMatchSnapshot();
  });
});
