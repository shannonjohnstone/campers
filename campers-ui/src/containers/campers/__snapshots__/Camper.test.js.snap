// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Camper component Component snapshot 1`] = `
<div>
  <table>
    <tbody>
      <tr>
        <td>
          1
        </td>
        <td>
          <img
            alt="Luann Ballard"
            src="https://www.tinygraphs.com/squares/859?theme=bythepool&numcolors=1&size=32&fmt=svg"
          />
        </td>
        <td>
          Luann Ballard
        </td>
        <td>
          55
        </td>
        <td>
          3543
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
